plugin/hpsa/hpsa.py

```diff
@@ -909,3 +933,223 @@ def volume_ident_led_clear(self, volume, flags=Client.FLAG_RSVD):
                 "Volume not found")
 
         return None
+
+    @_handle_errors
+    def batteries(self, search_key=None, search_value=None,
+                  flags=Client.FLAG_RSVD):
+        lsm_batteries = []
+        ctrl_all_show = self._sacli_exec(
+            ["ctrl", "all", "show", "config", "detail"])
+
+        for ctrl_name in ctrl_all_show.keys():
+            ctrl_data = ctrl_all_show[ctrl_name]
+            if 'Battery/Capacitor Count' not in ctrl_data or \
```

Comment:
I'm not opposed to that, but keep in mind it'll likely be a situation more like the bug I fixed the other day for cache modules. If a new string needed to be handled, we'd still need to handle the old string too, this would probably never go away.